src/arith/detect_linear_equation.cc

```diff
@@ -202,6 +203,9 @@ bool DetectClipBound(const PrimExpr& cond,
   } else if (const GENode* op = cond.as<GENode>()) {
     if (!op->a.dtype().is_int()) return false;
     canonical = op->a - op->b;
+  } else if (const EQNode* op = cond.as<EQNode>()) {
```

Comment:
For consistency, should this have the same check for `if(!op->a.dtype().is_int())` as the other cases?